Confidence/src/main/java/com/spotify/confidence/Confidence.kt

```diff
@@ -77,10 +80,23 @@ class Confidence internal constructor(
         return flagResolver.resolve(flags, getContext())
     }
 
-    suspend fun awaitReconciliation() {
-        if (currentFetchJob != null) {
-            currentFetchJob?.join()
-            activate()
+    suspend fun awaitReconciliation(timeoutMillis: Long = 5000) {
+        try {
+            withTimeout(timeoutMillis) {
+                while (currentFetchJob == null ||
+                    currentFetchJob?.isCompleted == true ||
+                    currentFetchJob?.isCancelled == true
```

Comment:
Maybe replace these with `currentFetchJob?.isActive != true`

Reply:
Mmm, perhaps. However I'm not certain having this is a great common timeout is great (imagine you call this function when we are actually NOT doing a network call, then it'll always just wait for timeoutMillis)... I might push an update.

Another approach would be to "delay" until `cache.flagEvaluation.context == getContext()` (but diffing them in some hashy way of course).